src/compiler/utilities.ts

```diff
@@ -84,6 +84,25 @@ namespace ts {
         return node.end - node.pos;
     }
 
+    export function mapIsEqualTo<T>(map1: Map<T>, map2: Map<T>): boolean {
+        if (!map1 || !map2) {
+            return map1 === map2;
+        }
+        return containsAll(map1, map2) && containsAll(map2, map1);

```

Comment:
it might but on our scenarios (maps with size < 10) I'll be more concerned about the cost of extra array allocations.